service/src/main/java/bio/terra/workspace/service/resource/controlled/ControlledResourceService.java

```diff
@@ -235,6 +236,56 @@ public String cloneGcsBucket(
     return jobBuilder.submit();
   }
 
+  public String cloneAzureContainer(
+      UUID sourceWorkspaceId,
+      UUID sourceResourceId,
+      UUID destinationWorkspaceId,
+      UUID destinationResourceId,
+      ApiJobControl jobControl,
+      AuthenticatedUserRequest userRequest,
+      @Nullable String destinationResourceName,
+      @Nullable String destinationDescription,
+      @Nullable String destinationContainerName,
+      @Nullable ApiCloningInstructionsEnum cloningInstructionsOverride) {
+    final ControlledResource sourceContainer =
+        getControlledResource(sourceWorkspaceId, sourceResourceId);
+
+    // Write access to the target workspace will be established in the create flight
+    final String jobDescription =
+        String.format(
+            "Clone controlled resource %s; id %s; name %s",
+            sourceContainer.getResourceType(),
+            sourceContainer.getResourceId(),
+            sourceContainer.getName());
+
+    // If TPS is enabled, then we want to merge policies when cloning a container
```

Comment:
Nit: could you replace TPS in the comment with Terra Policy Service? It took me some time to figure out what TPS stands for.

Reply:
Nit: would still appreciate this, if you end up needing to make more changes.